### DISCUSSION TITLE
Enable some allowed-by-default lints

### DISCUSSION THREAD

**Discussion by rusty-snake** [libseccomp-rs/libseccomp-rs#146] [Ideas]
Doc: https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html and https://rust-lang.github.io/rust-clippy/stable/index.html

### `warn(rust_2018_idioms)`

contains:

```
bare-trait-objects, unused-extern-crates, ellipsis-inclusive-range-patterns, elided-lifetimes-in-paths, explicit-outlives-requirements
```

Reason: Write more idiomatic Rust.

### `deny(missing_debug_implementations)`

Reason: enforce https://rust-lang.github.io/api-guidelines/debuggability.html

### `deny(missing_docs)`

Reason: Every public item should have documentation.

### `deny(unsafe_op_in_unsafe_fn)`

Reason: Safety: Limit `unsafe` LoC.

> Currently, an [unsafe fn](https://doc.rust-lang.org/reference/unsafe-functions.html) allows any unsafe operation within its body. However, this can increase the surface area of code that needs to be scrutinized for proper behavior. The unsafe block provides a convenient way to make it clear exactly which parts of the code are performing unsafe operations. In the future, it is desired to change it so that unsafe operations cannot be performed in an unsafe fn without an [unsafe](https://doc.rust-lang.org/reference/unsafety.html) block.

### `warn(clippy::inefficient_to_string)`

Reason: performance.

### `warn(clippy::string_to_string)` `warn(clippy::semicolon_if_nothing_returned)` `warn(clippy::clone_on_ref_ptr)`

Reason: readability.

---

Only `warn(rust_2018_idioms)` and `deny(unsafe_op_in_unsafe_fn) would actually require code changes.

**Comment by ManaSugi** [comment]
This is good. I agree to enable the lints.

**Comment by rusty-snake** [comment]
Note to self: Check if `clippy::unwrap_in_result` with `allow-unwrap-in-tests = true` can be used too.
